src/osdp_cp.c

```diff
@@ -1453,14 +1449,33 @@ static struct osdp *__cp_setup(int num_pd, const osdp_pd_info_t *info_list)
 
 	if (cp_detect_connection_topology(ctx)) {
```

Comment:
This has to be moved out to the caller. And both paths should call this method.

Reply:
done.